AEPCampaignClassic/Sources/CampaignClassicRegistrationManager.swift

```diff
@@ -0,0 +1,144 @@
+        var payload = String(format: CampaignClassicConstants.REGISTRATION_PAYLOAD_FORMAT,
+                             URLEncoder.encode(value: deviceToken),
+                             URLEncoder.encode(value: integrationKey),
+                             URLEncoder.encode(value: userKey),
+                             URLEncoder.encode(value: UIDevice.current.name),
+                             URLEncoder.encode(value: UIDevice.current.model),
+                             URLEncoder.encode(value: CampaignClassicConstants.REGISTER_PARAM_DEVICE_BRAND_APPLE),
+                             URLEncoder.encode(value: CampaignClassicConstants.REGISTER_PARAM_DEVICE_MANUFACTURER_APPLE),
+                             URLEncoder.encode(value: UIDevice.current.systemName),
+                             URLEncoder.encode(value: systemInfoService.getFormattedOperatingSystem()),
+                             URLEncoder.encode(value: systemInfoService.getFormattedLocale()))
+        payload.append(additionalParametersXML)
+
+        /// build url and header
+        let urlString = String(format: CampaignClassicConstants.REGISTRATION_API_URL_BASE, marketingServer)
+        let headers = buildHeaders(payload: payload)
+        guard let url = URL(string: urlString) else {
+            Log.debug(label: CampaignClassicConstants.LOG_TAG, "Device Registration failed, Invalid network request URL : \(urlString)")
+            return
+        }
+
+        /// crearte a network request
+        let request = NetworkRequest(url: url, httpMethod: .post, connectPayload: payload, httpHeaders: headers, connectTimeout: configuration.timeout, readTimeout: configuration.timeout)
+
+        Log.debug(label: CampaignClassicConstants.LOG_TAG, "Device Registration request initiated with \n URL : \(url.absoluteString) \n Headers: \(headers) \n Payload : \(payload)")
+
+        /// make the network request
+        ServiceProvider.shared.networkService.connectAsync(networkRequest: request, completionHandler: { connection in
+            if connection.responseCode != 200 {
+                Log.debug(label: CampaignClassicConstants.LOG_TAG, "Device Registration failed, Network Error. Response Code: \(String(describing: connection.responseCode)) URL : \(url.absoluteString)")
+                return
+            }
+
+            Log.debug(label: CampaignClassicConstants.LOG_TAG, "Device Registration success. URL : \(url.absoluteString)")
+            self.hashedRegistrationData = hashedData
+        })
+
+        return
+    }
+
+    /// Returns the network headers required for device registration request for the given payload
+    /// - Parameter payload : the registration request payload
+    private func buildHeaders(payload: String) -> [String: String] {
+        return [HttpConnectionConstants.Header.HTTP_HEADER_KEY_CONTENT_TYPE: HttpConnectionConstants.Header.HTTP_HEADER_CONTENT_TYPE_WWW_FORM_URLENCODED + ";" + CampaignClassicConstants.HEADER_CONTENT_TYPE_UTF8_CHARSET,
```

Comment:
nit: it's easier to read in-line dictionary construction when using one line per entry.  e.g. 
```
return [
    "key": "value",
    "anotherKey": "anotherValue"
]
```

Reply:
well actually it is that way right now. The value for one of the key is 
`HttpConnectionConstants.Header.HTTP_HEADER_CONTENT_TYPE_WWW_FORM_URLENCODED + ";" + CampaignClassicConstants.HEADER_CONTENT_TYPE_UTF8_CHARSET`

hence it seems out of place